src/main.js

```diff
@@ -98,13 +98,55 @@ class Entry {
    *@param {string|null} defaultValue given default value e.g. data
    *@param {string} comment given comment using the # symbol
    */
+
   constructor(name, type, choices, defaultValue, comment) {
     this.name = name;
     this.type = type;
     this.choices = choices;
     this.defaultValue = defaultValue;
     this.comment = comment;
   }
+
+  /** @method html
+   * @summary Returns the rendered HTML for the given entry
+   * @returns {promise} that resolves to string
+   */
+  html() {
+    const entry = [this];
+    return new Promise(function(resolve, reject) {
+      if (entry) {
```

Comment:
`entry` will always be true-ish here. I think what you mean to perform here is something like the following:

```js
const entry = outputHTML([this]);

if (entry) {
  resolve(entry);
else {
  reject('...');
}
```

Right?